### PR TITLE
1647 simulation output mappings

### PR DESCRIPTION
**Pull request by georgeDaskalakis**

### CODE REVIEW

File: src/OSPSuite.UI/Views/Charts/ChartEditorView.cs [hunk 1]
```diff
@@ -80,7 +80,7 @@ public ChartEditorView(IMenuBarItemToBarItemMapper barItemMapper, IImageListRetr
          repositoryItemCheckEditForUsedIn.ValueGrayed = null;
 
          _barEditItemLinkSimulationObserved.EditValue = false;
-         //_barEditItemLinkSimulationObserved.Visibility = BarItemVisibility.Never;
+         _barEditItemLinkSimulationObserved.Visibility = BarItemVisibility.Never;
```
Comment (msevestre) [inline src/OSPSuite.UI/Views/Charts/ChartEditorView.cs:83]: use SetlinkSimDataMenuItemVisisbility

Comment (msevestre) [reply src/OSPSuite.UI/Views/Charts/ChartEditorView.cs:83]: also rename SetlinkSimDataMenuItemVisisbility to remove the type
Are you using resharper? it should show you typo like this out of the box

File: tests/OSPSuite.Presentation.Tests/Presentation/ParameterIdentificationTimeProfileChartPresenterSpecs.cs [hunk 2]
```diff
@@ -186,13 +179,5 @@ public void should_hide_the_observed_data_curve_from_the_legend()
          var observedDataCurve = _timeProfileAnalysis.FindCurveWithSameData(_firstObservedData1.BaseGrid, _firstObservedData1);
          observedDataCurve.VisibleInLegend.ShouldBeFalse();
       }
-
-/*
```
Comment (msevestre) [inline tests/OSPSuite.Presentation.Tests/Presentation/ParameterIdentificationTimeProfileChartPresenterSpecs.cs]: Remove

**Review by msevestre — CHANGES_REQUESTED** [review]
It is still a draft. Please remove

File: src/OSPSuite.Core/Chart/ChartSettings.cs [hunk 3]
```diff
@@ -14,6 +14,14 @@ public enum LegendPositions
       BottomInside
    }
 
+   public enum GridGroupRowFormats
```
Comment (msevestre) [inline src/OSPSuite.Core/Chart/ChartSettings.cs:17]: Move out of ChartSettings. This is a concept that is relevant for all UI using a GridView

Comment (georgeDaskalakis) [reply src/OSPSuite.Core/Chart/ChartSettings.cs:17]: @msevestre this is not ready yet

File: src/OSPSuite.Core/Domain/ParameterIdentifications/OutputMapping.cs [hunk 4]
```diff
@@ -91,6 +91,35 @@ public bool DimensionsAreConsistent(DataRepository observedData)
          return false;
       }
 
+      public bool SimulationDimensionsAreConsistent(DataRepository observedData)
```
Comment (msevestre) [inline src/OSPSuite.Core/Domain/ParameterIdentifications/OutputMapping.cs:94]: I don't understand the name of this method. What does it do?

File: src/OSPSuite.Core/Domain/ParameterIdentifications/OutputMapping.cs [hunk 5]
```diff
@@ -91,6 +91,35 @@ public bool DimensionsAreConsistent(DataRepository observedData)
...
+
+         var observationColumn = observedData.FirstDataColumn();
+
+         if (Output.Dimension == null || observationColumn.Dimension == null)
+            return false;
+
+         if (Output.Dimension == observationColumn.Dimension)
+            return true;
+
+         if (Output.IsConcentration() && observationColumn.IsConcentration())
```
Comment (msevestre) [inline src/OSPSuite.Core/Domain/ParameterIdentifications/OutputMapping.cs:110]: hum... Aren't we missign other checks. Why is this required?

File: src/OSPSuite.Core/Domain/ParameterIdentifications/OutputMappings.cs [hunk 6]
```diff
@@ -46,6 +46,11 @@ public virtual IEnumerable<OutputMapping> OutputMappingsUsingDataRepository(Data
          return All.Where(x => x.UsesObservedData(dataRepository));
       }
 
+      public virtual IEnumerable<DataRepository> AllDataRepositoryMappedTo(string outputFullPath)
+      {
+         return All.Where(x => x.FullOutputPath == outputFullPath).Select(x => x.WeightedObservedData.ObservedData);
```
Comment (msevestre) [inline src/OSPSuite.Core/Domain/ParameterIdentifications/OutputMappings.cs:51]: Does it contain the SimulationName? Is this intended?
Also one is called FullOutputPath and the other outputFullPath. Let's make this consistent

**Comment by georgeDaskalakis** [comment]
@msevestre I am closing this PR for now, since the code is not yet ready. It was created for integration purposes in PK-Sim and MoBi.

File: src/OSPSuite.Presentation/Presenters/Charts/DataBrowserPresenter.cs [hunk 7]
```diff
@@ -150,6 +168,29 @@ public void UsedChangedFor(DataColumnDTO dataColumnDTO, bool used)
...
+      private List<DataColumnDTO> getLinkedObservedDataFromOutputPath(string updatedDataColumnName)
+      {
+         var linkedObservedDataRepositories = AllOutputMappings.AllDataRepositoryMappedTo(updatedDataColumnName);
+         var linkedObservedData = getDataColumnDTOsFromDatarepositories(linkedObservedDataRepositories);
+         return linkedObservedData;
+      }
+
+      private List<DataColumnDTO> getDataColumnDTOsFromDatarepositories(IEnumerable<DataRepository> linkedObservedDataRepositories)
+      {
+         var linkedObservedData = _dataColumnDTOCache.KeyValues.Where(x => linkedObservedDataRepositories.Any(y => y.Id == x.Key.Repository.Id))
```
Comment (msevestre) [inline src/OSPSuite.Presentation/Presenters/Charts/DataBrowserPresenter.cs:191]: FindById?

File: src/OSPSuite.Presentation/Presenters/SimulationOutputMappingPresenter.cs [hunk 8]
```diff
@@ -0,0 +1,239 @@
...
+         var newOutputMapping = new OutputMapping();
+         //_allOutputMappingDTOs.Add(mapFrom(newOutputMapping));  
+         OnStatusChanged();
+      }
+
+      private void updateOutputMappingList()
+      {
+         _listOfOutputMappingDTOs.Clear();
+
+         //KEEP SHIT SIMPLE: just add the observed data as mappings to the DTO list (hence also the simulation.OutputMappings)
```
Comment (msevestre) [inline src/OSPSuite.Presentation/Presenters/SimulationOutputMappingPresenter.cs:105]: REmove SHIT

**Review by msevestre — COMMENTED** [review]
I need you to walk me through the code. Too much stuff going on

**Comment by msevestre** [comment]
Why is this closed?

**Comment by msevestre** [comment]
oh  ok
